blocks/api/registration.js

```diff
@@ -141,6 +141,14 @@ export function registerBlockType( name, settings ) {
 	}
 
 	settings.icon = normalizeIconObject( settings.icon );
+	if ( ! isValidIcon( settings.icon.src ) ) {
+		console.error(
+			'The icon passed is invalid.' +
+			'The icon should be a string with a Slug of the DashIcon or an element (or function returning an element) if you choose to render your own SVG.' +
```

Comment:
Nitpicks:
 
- There will be no space between this sentence and the last with the string concatenation.
   - Same issue with line below.
- Slug is not a proper noun and does not need to be capitalized.
- "DashIcon" should be capitalized as "Dashicon"

Given how many forms this field supports, I might regret suggesting the guidance 😄 Or, viewed in a different light, maybe it helps highlight that we have too many options. We could also simplify this to a brief overview of supported types (string, element, component class, function) with an external link to the documentation for more information.

Reply:
I updated the string and linked to the docs. The previous text was a little bit big, this version is simpler. Space was added, it is a minor detail but would make the message hard to read.